client/chat/channel.jsx

```diff
@@ -68,15 +71,57 @@ const UserSublist = styled.ul`
 
 const UserListEntryItem = styled.li`
   ${userListRow};
+
+  &:hover {
+    background-color: rgba(255, 255, 255, 0.12);
+  }
 `
 
+@connect()
 class UserListEntry extends React.Component {
   static propTypes = {
     user: PropTypes.string.isRequired,
   }
 
+  state = {
+    userOverlayOpen: false,
+  }
+
+  _userEntryRef = React.createRef()
+
+  onOpenUserOverlay = () => {
+    this.setState({ userOverlayOpen: true })
+  }
+
+  onCloseUserOverlay = () => {
+    this.setState({ userOverlayOpen: false })
+  }
+
+  onWhisper = () => {
+    this.props.dispatch(navigateToWhisper(this.props.user))
+  }
+
+  renderUserOverlay() {
+    return (
+      <UserOverlay
+        open={this.state.userOverlayOpen}
+        onDismiss={this.onCloseUserOverlay}
+        anchor={this._userEntryRef.current}
+        user={this.props.user}>
+        <MenuItem text='Whisper' onClick={this.onWhisper} />
+      </UserOverlay>
+    )
+  }
+
   render() {
-    return <UserListEntryItem>{this.props.user}</UserListEntryItem>
+    return (
+      <>
+        <UserListEntryItem ref={this._userEntryRef} onClick={this.onOpenUserOverlay}>
+          {this.props.user}
+        </UserListEntryItem>
+        {this.renderUserOverlay()}
```

Comment:
I'm pretty sure we'll want to render the user overlay from the other parts of the application, and not just the chat channel. For example, from the whispers and lobbies. Assuming we'll want to render the _same_ user overlay in all of these places (and I can't see why not at the moment), then I think the logic that deals with rendering the user overlay should be moved up in hierarchy until it encompasses all of those places.

In our application, that's probably the `MainLayout` component. Keep in mind though, since the `MainLayout` is using `Route`s to render various stuff, that means you'll probably have to send certain props to the component that the `Route` is rendering. See `render` property in the react router documentation on how to do that easily: https://reactrouter.com/web/api/Route/render-func

Reply:
Actually, disregard this comment, it wasn't well thought through ^^.